tiledb/cloud/array.py

```diff
@@ -563,3 +602,150 @@ def apply(
         result_format_version=result_format_version,
         **kwargs,
     ).get()
+
+
+def exec_multi_array_udf_async(
+    func=None,
+    array_list=None,
+    name=None,
+    layout=None,
+    image_name=None,
+    http_compressor="deflate",
+    include_source_lines=True,
+    task_name=None,
+    result_format=rest_api.models.UDFResultType.NATIVE,
+    result_format_version=None,
+    **kwargs
+):
+    """
+    Apply a user defined function to multiple arrays
+    :param func: user function to run
+    :param array_details: dict from array name (tiledb format) to tuple (ranges, attributes)
+    :param layout: tiledb query layout
+    :param image_name: udf image name to use, useful for testing beta features
+    :param http_compressor: set http compressor for results
+    :param str task_name: optional name to assign the task for logging and audit purposes
+    :param UDFResultType result_format: result serialization format
+    :param str result_format_version: set a format version for cloudpickle or arrow IPC
+    :param kwargs: named arguments to pass to function
+    :return: UDFResult object which is a future containing the results of the UDF
+    >>> import numpy as np
+    >>> from tiledb.cloud import array
+    >>> import tiledb.cloud
+    >>> def median(numpy_ordered_dictionary):
+    ...     return np.median(numpy_ordered_dictionary["a"])
+
+    >>> dense_array = "tiledb://TileDB-Inc/quickstart_dense"
+    >>> sparse_array = "tiledb://TileDB-Inc/quickstart_sparse"
+    >>> array_details = dict([
+    >>>     (dense_array, ([(1, 4), (1, 4)], ["a"])),
+    >>>     (sparse_array, ([(1, 2), (1, 4)], ["a"]))
+    >>> ])
+
+    >>> res = array.exec_multi_array_udf(median, array_details)
+    >>> print("Median Multi UDF:\n{}\n".format(res))
+    """
+
+    api_instance = client.client.udf_api
+
+    if func is not None and not callable(func):
+        raise TypeError("func argument to `exec_multi_array_udf` must be callable!")
+    elif func is None and name is None or name == "":
+        raise TypeError("name argument to `apply` must be set if no function is passed")
+
+    pickledUDF = None
+    source_lines = None
+    if func is not None:
+        source_lines = utils.getsourcelines(func) if include_source_lines else None
+        pickledUDF = cloudpickle.dumps(func, protocol=udf.tiledb_cloud_protocol)
+        pickledUDF = base64.b64encode(pickledUDF).decode("ascii")
+
+    if array_list is None:
```

Comment:
Lets also check the type of the array_list passed

Reply:
The type of `array_list` is checked